From 6da8bf4a6399bee3412469f67952dd238a557da4 Mon Sep 17 00:00:00 2001
From: Hongzhang Yang <Hongzhang.Yang@freescale.com>
Date: Tue, 13 Aug 2013 11:29:31 +0800
Subject: [PATCH 239/280] ENGR00274902 VPU lib bug fixes

VPU lib v5.4.17

Bugs:
1) Size of /proc/cpuinfo may be larger than 1024 bytes
2) fclose(fp) is called twice if 1)
3) If user sets userQpMax=0, VPU lib ignores the max QP setting instead
   of setting max QP to 0

Signed-off-by: Hongzhang Yang <Hongzhang.Yang@freescale.com>
---
 vpu/vpu_io.c  | 3 +--
 vpu/vpu_lib.c | 2 +-
 vpu/vpu_lib.h | 2 +-
 3 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/vpu/vpu_io.c b/vpu/vpu_io.c
index 5f027c9..3861bf9 100644
--- a/vpu/vpu_io.c
+++ b/vpu/vpu_io.c
@@ -70,7 +70,7 @@ int isVpuInitialized(void)
 static int get_system_rev(void)
 {
 	FILE *fp;
-	char buf[1024];
+	char buf[2048];
 	int nread;
 	char *tmp, *rev;
 	int ret = -1;
@@ -84,7 +84,6 @@ static int get_system_rev(void)
 	nread = fread(buf, 1, sizeof(buf), fp);
 	fclose(fp);
 	if ((nread == 0) || (nread == sizeof(buf))) {
-		fclose(fp);
 		return ret;
 	}
 
diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index 4f89ef3..b49070c 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -1001,7 +1001,7 @@ RetCode vpu_EncGetInitialInfo(EncHandle handle, EncInitialInfo * info)
 	}
 
 	if (cpu_is_mx6x()) {
-		if(pEncInfo->openParam.userQpMax) {
+		if (pEncOP->userQpMaxEnable) {
 			data |= (1 << 6);
 			VpuWriteReg(CMD_ENC_SEQ_RC_QP_MIN_MAX, pEncInfo->openParam.userQpMax);
 		}
diff --git a/vpu/vpu_lib.h b/vpu/vpu_lib.h
index 9f57aea..d925bda 100644
--- a/vpu/vpu_lib.h
+++ b/vpu/vpu_lib.h
@@ -663,7 +663,7 @@ typedef struct vpu_versioninfo {
  * v4.2.2 [2008.09.03] support encoder on MX51
  * v4.0.2 [2008.08.21] add the IOClkGateSet() for power saving.
  */
-#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(5, 4, 16)
+#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(5, 4, 17)
 
 extern unsigned int system_rev;
 
-- 
1.8.0

